apps/api/src/view-models/buildViewModelForEntity/viewModelBuilders/buildPhotographViewModels.ts

```diff
@@ -0,0 +1,36 @@
+import { Photograph } from 'apps/api/src/domain/models/photograph/entities/photograph.entity';
+import IsPublished from 'apps/api/src/domain/repositories/specifications/isPublished';
+import { entityTypes } from 'apps/api/src/domain/types/entityTypes';
+import { isInternalError } from 'apps/api/src/lib/errors/InternalError';
+import { PhotographViewModel } from '../viewModels/photograph.view-model';
+import { ViewModelBuilderDependencies } from './types/ViewModelBuilderDependencies';
+import {
+    getDefaultViewModelBuilderOptions,
+    ViewModelBuilderOptions,
+} from './types/ViewModelBuilderOptions';
+
+const defaultOptions = getDefaultViewModelBuilderOptions();
+
+export default async (
+    { repositoryProvider, configService }: ViewModelBuilderDependencies,
+    optionOverrides: Partial<ViewModelBuilderOptions> = defaultOptions
+): Promise<PhotographViewModel[]> => {
+    const { shouldReturnUnpublishedEntities } = {
+        ...defaultOptions,
+        ...optionOverrides,
+    };
+
+    const baseAudioURL = configService.get<string>('BASE_AUDIO_URL');
+
+    const isPublishedSpecification = shouldReturnUnpublishedEntities ? null : new IsPublished(true);
+
+    const photographRepository = repositoryProvider.forEntity<Photograph>(entityTypes.photograph);
+
+    const searchResult = await photographRepository.fetchMany(isPublishedSpecification);
+
+    const allPhotographViewModels = searchResult
+        .filter((result): result is Photograph => !isInternalError(result))
+        .map((Photograph) => new PhotographViewModel(Photograph, baseAudioURL));
```

Comment:
This `baseAudioURL` got me confused, but I'm assuming it will eventually be renamed as something like `baseMediaURL`?

Reply:
GOOD POINT! I should do this **now**.

Reply:
done.